harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Harvester.java

```diff
@@ -389,13 +389,6 @@ private void searchAndAlign(CswServer server, Search s, Set<String> uuids,
                 break;
             }
 
-            // Another way to escape from an infinite loop
-
-            if (returnedCount == 0) {
-                log.warning("Forcing harvest end since numberOfRecordsReturned = 0");
-                break;
-            }
-
```

Comment:
Maybe also change 
```
start = nextRecord;
```
line 401.

@josegar74 any opinion?

Reply:
I'm missing something like
```java
            if (nextRecord < start) {
                log.warning(String.format("Forcing harvest end since nextRecord < start (nextRecord = %d, start = %d", nextRecord, start));
                break;
            }
```